File: Java/app/src/main/java/com/taboola/sdk4example/sdk_classic/MetaClassicUnitFragment.java
package com.taboola.sdk4example.sdk_classic;

import static com.taboola.sdk4example.Const.META_FEED_MODE;
import static com.taboola.sdk4example.Const.META_FEED_PLACEMENT_NAME;
import static com.taboola.sdk4example.Const.META_WIDGET_MODE;
import static com.taboola.sdk4example.Const.META_WIDGET_PLACEMENT_NAME;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;

import com.facebook.ads.NativeAdLayout;
import com.taboola.android.MetaPlacementProperties;
import com.taboola.android.TBLClassicPage;
import com.taboola.android.TBLMetaClassicUnit;
import com.taboola.android.TBLPublisherInfo;
import com.taboola.android.Taboola;
import com.taboola.android.annotations.TBL_PLACEMENT_TYPE;
import com.taboola.android.listeners.TBLClassicListener;
import com.taboola.sdk4example.Const;
import com.taboola.sdk4example.MetaConst;
import com.taboola.sdk4example.R;
import com.taboola.sdk4example.tabs.BaseTaboolaFragment;

import java.util.HashMap;

public class MetaClassicUnitFragment extends BaseTaboolaFragment {

    private static final String TAG = MetaClassicUnitFragment.class.getSimpleName();

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        Taboola.init(new TBLPublisherInfo(MetaConst.META_PUBLISHER_NAME));
        View rootView = inflater.inflate(R.layout.fragment_meta_ad_classic_unit, null);
        NativeAdLayout adContainerTop = rootView.findViewById(R.id.native_ad_container_top);

        Taboola.setGlobalExtraProperties(new HashMap<String, String>() {{
            put(MetaConst.AUDIENCE_NETWORK_APPLICATION_ID_KEY, MetaConst.AUDIENCE_NETWORK_APP_ID);
            put(MetaConst.ENABLE_META_DEMAND_DEBUG_KEY, "true");
        }});

        setupAndLoadTaboolaAd(adContainerTop);
        return rootView;
    }


    private void setupAndLoadTaboolaAd(NativeAdLayout adContainer) {
        TBLClassicPage tblClassicPage = Taboola.getClassicPage(Const.PAGE_URL, Const.PAGE_TYPE);
        MetaPlacementProperties metaPlacementProperties = new MetaPlacementProperties(META_WIDGET_PLACEMENT_NAME, META_WIDGET_MODE);
        TBLMetaClassicUnit tblMetaClassicUnit = tblClassicPage.buildWithMeta(getContext(), META_FEED_PLACEMENT_NAME, META_FEED_MODE, TBL_PLACEMENT_TYPE.PAGE_MIDDLE, metaPlacementProperties, new TBLClassicListener() {
            @Override
            public boolean onItemClick(String placementName, String itemId, String clickUrl, boolean isOrganic, String customData) {
                Log.d(TAG, "onItemClick");
                return super.onItemClick(placementName, itemId, clickUrl, isOrganic, customData);
            }

            @Override
            public void onAdReceiveSuccess() {
                super.onAdReceiveSuccess();
                Log.d(TAG, "onAdReceiveSuccess");
            }

            @Override
            public void onAdReceiveFail(String error) {
                super.onAdReceiveFail(error);
                Log.d(TAG, "onAdReceiveFail " + error);
            }

            @Override
            public void onResize(int height) {
                super.onResize(height);
                Log.d(TAG, "onResize");
            }
        });
        tblMetaClassicUnit.setMetaAdTypeForDebug(MetaConst.TEST_LAYOUT_AD_TYPE);
        tblMetaClassicUnit.setUnitExtraProperties(new HashMap<String, String>() {{
            put(MetaConst.AUDIENCE_NETWORK_PLACEMENT_ID_KEY, MetaConst.AUDIENCE_NETWORK_PLACEMENT_ID);
        }});

        tblMetaClassicUnit.setMetaNativeUI(MetaConst.DEFAULT_LAYOUT_KEY);
        adContainer.addView(tblMetaClassicUnit);
        tblMetaClassicUnit.fetchContent();
    }

}

File: Kotlin/app/src/main/java/com/taboola/kotlin/examples/screens/classic/MetaClassicUnitFragment.kt
package com.taboola.kotlin.examples.screens.classic

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.facebook.ads.NativeAdLayout
import com.taboola.android.MetaPlacementProperties
import com.taboola.android.TBLPublisherInfo
import com.taboola.android.Taboola
import com.taboola.android.annotations.TBL_PLACEMENT_TYPE
import com.taboola.android.listeners.TBLClassicListener
import com.taboola.kotlin.examples.MetaConst
import com.taboola.kotlin.examples.PlacementInfo
import com.taboola.kotlin.examples.R

class MetaClassicUnitFragment : Fragment() {

    private var rootView: View? = null
    private var adContainerTop: NativeAdLayout? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        Taboola.init(TBLPublisherInfo(MetaConst.META_PUBLISHER_NAME))
        rootView = inflater.inflate(R.layout.fragment_meta_ad_classic_unit, null)
        adContainerTop = rootView!!.findViewById<NativeAdLayout>(R.id.native_ad_container_top)
        Taboola.setGlobalExtraProperties(object : HashMap<String?, String?>() {
            init {
                put(MetaConst.AUDIENCE_NETWORK_APPLICATION_ID_KEY, MetaConst.AUDIENCE_NETWORK_APP_ID)
                put(MetaConst.ENABLE_META_DEMAND_DEBUG_KEY, "true")
            }
        })
        adContainerTop?.let { setupAndLoadTaboolaAd(it) }
        return rootView
    }


    private fun setupAndLoadTaboolaAd(adContainer: NativeAdLayout) {
        val widgetProperties = PlacementInfo.metaWidgetProperties()
        val feedProperties = PlacementInfo.metaFeedProperties()
        val tblClassicPage = Taboola.getClassicPage(feedProperties.pageUrl, feedProperties.pageType)
        val metaPlacementProperties =
            MetaPlacementProperties(widgetProperties.placementName, widgetProperties.mode)
        val tblMetaClassicUnit = tblClassicPage.buildWithMeta(
            context,
            feedProperties.placementName,
            feedProperties.mode,
            TBL_PLACEMENT_TYPE.PAGE_MIDDLE,
            metaPlacementProperties,
            object : TBLClassicListener() {})
        tblMetaClassicUnit.setMetaAdTypeForDebug(MetaConst.TEST_LAYOUT_TYPE)
        tblMetaClassicUnit.setUnitExtraProperties(object : HashMap<String?, String?>() {
            init {
                put(MetaConst.AUDIENCE_NETWORK_PLACEMENT_ID_KEY, MetaConst.AUDIENCE_NETWORK_PLACEMENT_ID)
            }
        })
        tblMetaClassicUnit.setMetaNativeUI(MetaConst.DEFAULT_LAYOUT_KEY)
        adContainer.addView(tblMetaClassicUnit)
        tblMetaClassicUnit.fetchContent()
    }
}


